public/app/percona/settings/components/Communication/Email/Email.tsx

```diff
@@ -128,18 +146,35 @@ export const Email: FC<EmailProps> = ({ updateSettings, settings }) => {
             />
             <CheckboxField name="requireTls" label="Require TLS" />
 
-            <Button
-              className={settingsStyles.actionButton}
-              type="submit"
-              disabled={!valid || pristine || loading}
-              data-testid="email-settings-submit-button"
-            >
-              {loading && <Spinner />}
-              {Messages.actionButton}
-            </Button>
+            {testRef.current &&
+              createPortal(
+                <TestEmailSettings
+                  onInput={(email) => (testEmailRef.current = email)}
+                  onTest={(email) => handleTestClick(values, email)}
+                  initialValue={settings.test_email}
+                />,
+                testRef.current
+              )}
+
+            {applyRef.current &&
+              createPortal(
+                <Button
+                  className={settingsStyles.actionButton}
+                  type="submit"
+                  disabled={!valid || pristine || loading}
+                  data-testid="email-settings-submit-button"
+                  onClick={handleSubmit}
+                >
+                  {loading && <Spinner />}
+                  {Messages.actionButton}
+                </Button>,
+                applyRef.current
+              )}
           </form>
         )}
       />
+      <div ref={(e) => (testRef.current = e)}></div>
```

Comment:
```suggestion
      <div ref={testRef}></div>
```

Isn't this the same?

Reply:
Well, the problem is that, initially, the ref will be null. We could either use `useState` to observe changes on `testRef` and make the component re-render, or just simply use a callback ref: https://reactjs.org/docs/refs-and-the-dom.html#callback-refs
Doing this, by the time the component mounts, it's guaranteed our ref is updated, as stated in the docs.